integration/e2ecortex/client.go

```diff
@@ -627,6 +629,12 @@ func (c *Client) GetPrometheusRules(filter RuleFilter) ([]*ruler.RuleGroup, erro
 	addQueryParams(urlValues, "rule_group[]", filter.RuleGroupNames...)
 	addQueryParams(urlValues, "type", filter.RuleType)
 	addQueryParams(urlValues, "exclude_alerts", filter.ExcludeAlerts)
+	if filter.MaxRuleGroup > 0 {
+		addQueryParams(urlValues, "group_limit", strconv.Itoa(filter.MaxRuleGroup))
+	}
+	if filter.NextToken != "" {
+		addQueryParams(urlValues, "group_next_token", filter.NextToken)
+	}
```

Comment:
Nit: If we just add `group_next_token=""` what will happen? It should still be a non paginated request?
Maybe we can just remove the condition check here

Reply:
To keep consistency with upstream, we should fail if `group_next_token` was provided but `group_limit` was not.

Reply:
Never mind, I see the validation in the API.